ui/src/views/user/survey-edit/SurveyEditView.vue

```diff
@@ -240,22 +249,60 @@ export default {
       return this.formData;
     },
     onSaveAsDraftOptionClick() {
+      console.log("save as draft");
       const finalOutput = this.getData();
       console.log(JSON.stringify(finalOutput));
 
-      this.$notify.toast("Survey has been successfully saved");
+      if(finalOutput.config.startDate == "" || finalOutput.config.endDate == "") {
+        this.$notify.toast("Please give survey's start date and end date!");
+      } else {
+        if (this.currentSurveyId == "new") {
+          userAddSurvey(finalOutput)
+          .then(() => {
+            this.$notify.toast("Your survey has been saved!");
+            this.$router.push("/user/surveys/");
+          }).catch((error) => {
+            this.$notify.toast(error["message"]);
+          });
+        } else {
+          finalOutput.config.id = this.currentSurveyId;
+          userEditSurvey(finalOutput)
+          .then(() => {
+            this.$notify.toast("Your survey has been saved!");
+            window.location.reload();
+          }).catch((error) => {
+            this.$notify.toast(error["message"]);
+          });
+        }
+      }
     },
     onSaveAndPublishOptionClick() {
       console.log("save and publish");
       const finalOutput = this.getData();
       console.log(JSON.stringify(finalOutput));
 
-      // call api
-      // save
-      // go to detail page
-
-      this.$notify.toast("Survey has been successfully published");
-      this.$router.push("/user/surveys/1");
+      if(finalOutput.config.startDate == "" || finalOutput.config.endDate == "") {
```

Comment:
I see that this function is similar with the save as draft one, why not combining it as one? you can change the function params to make it more dynamic

Reply:
these two functions were already in the code when I started working; later they will do slightly different things (the saveAndPublish will call BE api for sending email invitations -- this api currently is still unavailable)

you are right; i think i can combine them into one function with a parameter to choose between save draft/publish

Reply:
Yeah, the onClick functions should be there, but inside it they can call different functions XD